nether/portal.lua

```diff
@@ -176,6 +199,62 @@ minetest.register_chatcommand("from_hell", {
 	end
 })
 
+-- Useful for debugging Nether player state tracking. Written by Deathwing777
+minetest.register_chatcommand("in_hell", {
+	params = "[<player_name>]",
+	description = "Is the player in hell?",
+	func = function(name, pname)
+		if not minetest.check_player_privs(name, {nether=true}) then
+			return false,
+				"You need the nether priv to execute this chatcommand."
+		end
+		if not player_exists(pname) then
+			pname = name
+		end
+		local player = minetest.get_player_by_name(pname)
+		if not player then
+			return false, "Something went wrong."
+		end
+		
+		status = pname.." is in the "
+		if nether.is_player_in_nether(player) then
+			status = status.."NETHER!"
+		else
+			status = status.."OVERWORLD!"
+		end
+		
+		return true, status
+	end
+})
+
+minetest.register_chatcommand("update_hells_registry", {
+	params = "[<player_name>]",
+	description = "Update player state if they got to or from the nether in another way.",
+	func = function(name, pname)
+		if not minetest.check_player_privs(name, {nether=true}) then
+			return false,
+				"You need the nether priv to execute this chatcommand."
+		end
+		if not player_exists(pname) then
+			pname = name
+		end
+		local player = minetest.get_player_by_name(pname)
+		if not player then
+			return false, "Something went wrong."
+		end
+		
+		nether.external_nether_teleport(player)
+		status = pname.." is in the "
+		if nether.is_player_in_nether(player) then
+			status = status.."NETHER!"
+		else
+			status = status.."OVERWORLD!"
+		end
+		
```

Comment:
Style:
Here are two tabs in the empty line.
This is not the only line with trailing whitespace:

>    portal.lua:218:1: line contains only whitespace
    portal.lua:225:1: line contains only whitespace
    portal.lua:245:1: line contains only whitespace
    portal.lua:253:1: line contains only whitespace

Reply:
Python habits.... Odd that lua would complain about whitespace only lines rather than just silently stripping them out.

Reply:
Lua doesn't complain but [luacheck](https://github.com/lunarmodules/luacheck/) does, which is a static analyzer and linter for Lua code.